hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -1446,6 +1450,41 @@ public void removeAll(Set<Data> keys, int completionId) {
         return keysLoaded;
     }
 
+    @Override
+    public CacheRecord merge(SplitBrainMergeEntryView<Data, Data> mergingEntry, SplitBrainMergePolicy mergePolicy) {
```

Comment:
can we unify new and old merge methods?

Reply:
I found no useful way to do this. The common parts are local variables (which you cannot pull out in shared methods). And the shared logic differs in the type of classes and parameters.

Reply:
main skeleton of the methods seems identical, mergePolicy and entryView parts are different, maybe we can try to unify it later.

Reply:
The whole split-brain logic is very, very similar between most data structures. But the differences are either in the details of the data structure (how to retrieve the internal records/items) and of course in the classes, which are different everywhere.

Matko had the same thoughts on unifying logic between the data structures, but so far I found no sensible way of doing so. But we should definitely have a look once everything is merged. Maybe there are parts which can be unified and cleaned up. At least the interfaces are the same now for all new merge policies, which could help a lot.